processing/src/main/java/org/apache/druid/segment/join/table/BroadcastSegmentIndexedTable.java

```diff
@@ -254,8 +254,11 @@ public void close()
   }
 
   @Override
-  public Optional<Closeable> acquireReferences()
+  public Optional<Closeable> acquireReference()
   {
-    return Optional.empty();
+    // this is wrong, we should have a reference to the reference counted segment instead of the direct segment so that
```

Comment:
was it wrong prior to this patch? what issues will the wrongness cause?

Reply:
wrong prior to this patch, the issues that could happen is that because the joinable doesn't grab a reference to the underlying segment, there is no guard on the segment being dropped during processing. This is mainly for integration tests and isn't really documented so is probably not a huge problem in practice, as a broadcast segment it will only be dropped if it is updated with a newer version.